### ISSUE TITLE
release-please failed: Error adding to tree

### ISSUE THREAD

**Issue by bartutiryakioglu** [google-github-actions/release-please-action#694]
### TL;DR

PR not created when a commit received

### Expected behavior

PR to be created

### Observed behavior

Action stops running with the following error:
release-please failed: Error adding to tree: 93332e4552b999bf6d02e56f5044507a77ea92ca

----------------------------------------------------------------------------------------------
93332e4552b999bf6d02e56f5044507a77ea92ca is the latest commit for the default-branch

### Action YAML

```yaml
on:
  push:
    branches:
      - features
name: release-please
jobs:
  release-please:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - uses: google-github-actions/release-please-action@v3
        with:
          default-branch: features
          release-type: simple
          include-v-in-tag: false
          changelog-types: '[{"type":"feat","section":":sparkles: Features","hidden":false},{"type":"fix","section":":bug: Bug Fixes","hidden":false},{"type":"plugin","section":":package: Plugin Changes","hidden":false},{"type":"perf","section":":zap: Performance Improvements","hidden":false},{"type":"add","section":":hammer: Improvements","hidden":false},{"type":"revert","section":":rewind: Commit Reverts","hidden":false},{"type":"doc","section":"📚 Asset & Documentation Changes","hidden":false},{"type":"test","section":":white_check_mark: Tests","hidden":false},{"type":"ci","section":"🎡 CI","hidden":false},{"type":"refactor","section":"💡 Refactors","hidden":false},{"type":"release","section":"🏹  Releases","hidden":false},{"type":"style","section":"💄 Code Style Changes","hidden":false}]'
```

### Log output

```text
Run google-github-actions/release-please-action@v3.7.2
✔ Looking for latest release on branch: features with prefix: 
❯ Fetching merge commits on branch features with cursor: undefined
❯ Found latest release pull request: 511 version: 1.14.3
❯ Fetching releases with cursor undefined
❯ found release for  Version {
  major: 1,
  minor: 14,
  patch: 3,
  preRelease: undefined,
  build: undefined
}
❯ found release for  Version {
  major: 1,
  minor: 14,
  patch: 2,
  preRelease: undefined,
  build: undefined
}
❯ SHA not found in recent commits to branch features, skipping
❯ found release for  Version {
  major: 1,
  minor: 14,
  patch: 1,
  preRelease: undefined,
  build: undefined
}
❯ SHA not found in recent commits to branch features, skipping
❯ found release for  Version {
  major: 1,
  minor: 14,
  patch: 0,
  preRelease: undefined,
  build: undefined
}
❯ Found 2 files
❯ Backfilling file list for commit: 9e9fb893033c062fabfc10b444b049a39dbd25e6
❯ Found 1 files
❯ Backfilling file list for commit: 7ad9172e126759acba06652c1c29ed802b9c7f13
❯ Found 3 files
❯ Backfilling file list for commit: 62de7d6e176969e40da9bdc121dd9e43ff11ef1b
❯ Found 1 files
❯ Backfilling file list for commit: 36ba924debcac44c8eed981116fb1cedbbea379e
❯ Found 1 files
✔ PR #512 has many files, backfilling
❯ Backfilling file list for commit: 21b720148a94288ede812027bca9cc2fc425aab6
⚠ Found 3000 files. This may not include all the files.
❯ Backfilling file list for commit: b900986650e2d562cd205aeceb133a994ff2b4c2
❯ Found 16 files
❯ Backfilling file list for commit: 43a59640e987a77766b5421c157371d29bf1262d
❯ Found 1 files
❯ Backfilling file list for commit: 0ddb0f1eb335ef5fb0363d71dac02121aba2e8f1
❯ Found 1 files
❯ Backfilling file list for commit: 2830afe407cc327e927623932c5b2aef54cc04f5
❯ Found 3 files
❯ Backfilling file list for commit: f40a395c45b743717b8726869dc7ff5bad61419a
❯ Found 2 files
❯ Backfilling file list for commit: 618868ffd25a028cb9537edcf26f090e2b495094
❯ Found 2 files
❯ Backfilling file list for commit: 495c809e2ae6b7941f62fd19613d374edd88f038
❯ Found 3 files
❯ Backfilling file list for commit: 5dc0b1af1defbfcbfee6c39e9767dc089de0d365
❯ Found 1 files
❯ Backfilling file list for commit: 0ea33afcff39bda4417c74954f09047be63fda31
❯ Found 1 files
❯ Backfilling file list for commit: c5fd9aaa47fc8443a901baa0b8ec443612de41ba
❯ Found 6 files
❯ Backfilling file list for commit: f1ebf8947916aca807b015c1c6342e1d2fd59b9d
❯ Found 1 files
❯ Backfilling file list for commit: a69b7c30143977d92a6936cb416caf46531eeade
❯ Found 1 files
✔ Splitting 3 commits by path
✔ Building candidate release pull request for path: .
❯ type: simple
❯ targetBranch: features
❯ commits: 3
✔ Considering: 3 commits
❯ component: 
❯ pull request title pattern: undefined
✔ Looking for open release pull requests
✔ found 0 open release pull requests.
✔ Looking for snoozed release pull requests
✔ found 0 snoozed release pull requests.
❯ Fetching CHANGELOG.md from branch features
❯ Fetching version.txt from branch features
⚠ file version.txt did not exist
❯ Fetching .release-please-manifest.json from branch features
⚠ file .release-please-manifest.json did not exist
✔ Starting GitHub PR workflow...
✔ Successfully found branch HEAD sha "93332e4552b999bf6d02e56f5044507a77ea92ca".
✔ Skipping branch creation step...
✔ Got the latest commit tree
Error: release-please failed: Error adding to tree: 93332e4552b999bf6d02e56f5044507a77ea92ca
```

### Additional information

We were using this action without any problems for at least 3 months. But we had this problem since yesterday. Can't find anything on the internet to fix this error, hope you could help.

**Comment by chingor13** [comment]
I suggest trying to debug with the underlying `release-please` library's CLI: https://github.com/googleapis/release-please/blob/main/docs/troubleshooting.md (specifically with the `--trace`) option.

**Comment by alexisponcet** [comment]
Same problem with the manifest command. However, the release-please cli does work. It fails while trying to create PR requests.

**Comment by jwenz723** [comment]
I'm also experiencing this error. Here are the logs from my github actions run:

<pre>
✔ Building pull requests
✔ Building strategies by path
❯ .: simple
✔ Collecting release commit SHAs
❯ release search depth: 400
❯ Fetching releases with cursor undefined
❯ Found release for path ., v0.8.0
❯ release for path: ., version: 0.8.0, sha: 446f84cb5d32bf81bfc7e6f69e7a52a681e9e7ee
✔ Collecting commits since all latest releases
❯ commit search depth: 500
❯ Set(1) { '446f84cb5d32bf81bfc7e6f69e7a52a681e9e7ee' }
❯ Fetching merge commits on branch main with cursor: undefined
❯ Backfilling file list for commit: 9372c76061ba7957c7f7f556bc3039d383298266
❯ Found 4 files
❯ Backfilling file list for commit: 24dac357cf170e34[658](https://github.com/inContact/acddevops-reusable-workflows/actions/runs/4358230947/jobs/7618535898#step:3:659)6ec444e9ee7aff69[685](https://github.com/inContact/acddevops-reusable-workflows/actions/runs/4358230947/jobs/7618535898#step:3:686)43
❯ Found 1 files
❯ Backfilling file list for commit: fdd5b2f4104a652c389e462587d3a54abe[700](https://github.com/inContact/acddevops-reusable-workflows/actions/runs/4358230947/jobs/7618535898#step:3:701)46f
❯ Found 1 files
❯ Backfilling file list for commit: 71b837d7be17f735aaa8c3033a86e391ec5b90bf
❯ Found 1 files
❯ Backfilling file list for commit: 9486f516bef41efcf2224b235a36a8634de3fd05
❯ Found 1 files
❯ Backfilling file list for commit: 27312b361616e1f0ce81eb43f40065e1852df020
❯ Found 1 files
❯ Backfilling file list for commit: 88fa79073cba53923e722a94317ca2e57dba1621
❯ Found 3 files
❯ Backfilling file list for commit: 314debe51310d9443b58202913edaf69ec4a36ef
❯ Found 2 files
❯ Backfilling file list for commit: e0a88ca4fdf7eeaa61247ec2fac8571aa0223876
❯ Found 9 files
✔ Splitting 9 commits by path
✔ Building candidate release pull request for path: .
❯ type: simple
❯ targetBranch: main
❯ commits: 9
✔ Considering: 9 commits
❯ component: 
❯ pull request title pattern: undefined
✔ Looking for open release pull requests
✔ found 0 open release pull requests.
✔ Looking for snoozed release pull requests
✔ found 0 snoozed release pull requests.
❯ Fetching CHANGELOG.md from branch main
❯ Fetching version.txt from branch main
❯ Fetching .github/actions/dockerfile/default/action.yml from branch main
❯ Fetching .github/actions/dotnet/build/action.yml from branch main
❯ Fetching .github/actions/go/build/action.yml from branch main
❯ Fetching .github/actions/java/build/action.yml from branch main
❯ Fetching .github/workflows/code-build-and-deploy.yml from branch main
❯ Fetching .release-please-manifest.json from branch main
⚠ file .release-please-manifest.json did not exist
✔ Starting GitHub PR workflow...
✔ Successfully found branch HEAD sha "9372c76061ba7957c7f7f556bc3039d383298266".
✔ Skipping branch creation step...
✔ Got the latest commit tree
Error: release-please failed: Error adding to tree: 9372c76061ba7957c7f7f556bc3039d383298266
</pre>

Locally I executed:

```bash
release-please release-pr --token $GITHUB_TOKEN --repo-url=my-repo --debug --dry-run
```

and got the following output:

<pre>
ConfigurationError: base (my-repo): Missing required manifest config: release-please-config.json
    at fetchManifestConfig (/home/linuxbrew/.linuxbrew/lib/node_modules/release-please/build/src/manifest.js:837:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async parseConfig (/home/linuxbrew/.linuxbrew/lib/node_modules/release-please/build/src/manifest.js:790:20)
    at async Promise.all (index 0)
    at async Manifest.fromManifest (/home/linuxbrew/.linuxbrew/lib/node_modules/release-please/build/src/manifest.js:118:93)
    at async Object.handler (/home/linuxbrew/.linuxbrew/lib/node_modules/release-please/build/src/bin/release-please.js:309:24) {
  releaserName: 'base',
  repository: 'my-repo'
}
</pre>


Next, I added a `release-please-config.json` and `.release-please-manifest.json` file to my repo and updated my GitHub Actions workflow to the following:

```yaml
      - uses: google-github-actions/release-please-action@v3
        id: release
        with:
          command: manifest
```

When the workflow ran it returned the same error:

```bash
Error: release-please failed: Error adding to tree: cad1a4570893fa785e7f10698e4c48468218b3f6
```

Nex, I ran the `release-please` cli locally with the command:

```bash
release-please release-pr --token $GITHUB_TOKEN --repo-url=my-repo --debug --dry-run
```

This command succeeded. So I'm a little bit lost on what to do since the release-please CLI is working but the release-please-action is not.

Also, the release-please-action doesn't have the ability to enable additional debug logging, so I'm not sure what to do next to diagnose this issue.

**Comment by jwenz723** [comment]
After further diagnosing I've determined that if I include the `packages[*].extra-files` array then the `Error: release-please failed: Error adding to tree` error occurs. If I remove the `extra-files` array then the error no longer occurs.

I'm not familiar with the code within this repo, but I think this can likely help someone else to resolve this issue.

**Comment by MxNxPx** [comment]
Another data point...

Not sure if this is intentional, but have confirmed if using the `release-please-config.json`, and having the `extra-files` config section reference a file under the `.github/workflows` directory, release-please throws the same error as others have commented here:

```console
✔ Got the latest commit tree
Error: release-please failed: Error adding to tree: <<GIT_SHA>>
```

**Comment by dsiebel** [comment]
I am currently experiencing a similar problem ("error: adding to tree", `exclude-paths` instead of `extra-files`, described in #704 ).
I am wondering if the issues are related.
The current theory is that the path `.github/workflows` has some sort of special meaning.
Ring any bells?

**Comment by foxted** [comment]
The `.github/workflows` folder indeed requires another permission. I tried using a custom Personal Access Token with the following permissions and it worked: 

![Screen Shot 2023-09-21 at 3 20 18 AM](https://github.com/google-github-actions/release-please-action/assets/1385160/7bb7898d-3806-4caa-8cd5-9e2078a34331)

**Comment by Jmainguy** [comment]
Was getting the same issue trying to update a file in .github/ directory, adding the permissions fixed it. Thank you @foxted

**Comment by TimothyJones** [comment]
I was getting this issue with a fine grained access token - I couldn't find a set of permissions that worked, even if I matched the ones above.

Switching to a classic access token with the permissions in @foxted 's comment fixed it.

**Comment by okozachenko1203** [comment]
After we migrated to Github enterprise, this issue started to occur in our pipelines also. I want to share how we fixed it.

- Grant more permissions to `GITHUB_TOKEN`
```yaml
permissions:
  contents: write
  pull-requests: write
```
- Allow GitHub Actions to create and approve pull requests. You can check it in your org settings or Github enterprise settings.
